Hymns/Common/SoundCloudWebView.swift

```diff
@@ -0,0 +1,65 @@
+import SwiftUI
+import WebKit
+
+struct SoundCloudWebView: UIViewRepresentable {
+    @Binding var isLoading: Bool  //Used for activity indicator
+    var url: URL
+    var loadStatusChanged: ((Bool, Error?) -> Void)?
+
+    func makeCoordinator() -> SoundCloudWebView.Coordinator {
+        Coordinator(self)
+    }
+
+    func makeUIView(context: Context) -> WKWebView {
+        let view = WKWebView()
+        view.navigationDelegate = context.coordinator
+        view.load(URLRequest(url: url))
+        return view
+    }
+
+    func updateUIView(_ uiView: WKWebView, context: Context) {
+    }
+
+    func onLoadStatusChanged(perform: ((Bool, Error?) -> Void)?) -> some View {
+        var copy = self
+        copy.loadStatusChanged = perform
+        return copy
+    }
+
+    class Coordinator: NSObject, WKNavigationDelegate {
+        let parent: SoundCloudWebView
+
+        init(_ parent: SoundCloudWebView) {
+            self.parent = parent
+        }
+
+        func webView(_ webView: WKWebView, didCommit navigation: WKNavigation!) {
+            parent.loadStatusChanged?(true, nil)
+        }
+
+        func webView(_ webView: WKWebView, didFinish navigation: WKNavigation!) {
+
+            parent.isLoading = false  //Used for activity indicator
+
+            webView.evaluateJavaScript("document.querySelector('.upsellBanner__appButton').innerHTML = 'SoundCloud Results';", completionHandler: { (_, _) -> Void in
```

Comment:
If you're going to change the top bar, can you remove the click listener too? Now it still looks like clicking it brings the popup up

Reply:
I can try. I guess the question is should we even change it anymore. I think the reason I was looking to change it was that we didn't think the button worked. But after testing on a real device, it does seem like the button works and lets you got to the app. What do you think? Better to change and disable or leave in?

Reply:
If it works well, we should leave it. But I don't think it works well though. The "Download Soundcloud" button doesn't work for me. Does it work for you?